src/linker/Linker.Steps/ResolveFromAssemblyStep.cs

```diff
@@ -64,15 +64,12 @@ protected override void Process ()
 				Context.Resolver.CacheAssembly (_assembly);
 
 			var ignoreUnresolved = Context.Resolver.IgnoreUnresolved;
-			if (_rootVisibility == RootVisibility.PublicAndFamily) {
-				Context.Resolver.IgnoreUnresolved = false;
-			}
-
+			Context.Resolver.IgnoreUnresolved = false;
 			AssemblyDefinition assembly = _assembly ?? Context.Resolve (_file);
```

Comment:
This will treat `-a` the same as we treat `-r` today. There may be more room for improvement here. Ideally we would have a `Resolve` method that lets the caller decide what to do on resolution failure.

Reply:
I find this weird: the IgnoreUnresolved is a user facing setting, and we're deliberately ignore it here. I guess I would have to understand why we're doing this. That said, if we get a desired behavior... I'm fine with it.

Reply:
Agreed. There was some discussion of it in https://github.com/mono/linker/pull/929. I looked into making `Resolve` more flexible, but it looks to be more effort than it's worth right now.